refactor: Change blocked RPC to non-blocked RPC

ajblane opened the pull request:
In the case of all remote workers which are not workable, blocked PRCs
not get RPC results until remote workers are workable. Therefore,
remote interface needs to change blocked RPC to non-blocked RPC with
10s. When time-out occurs, remote interface goes to run local dcurl.

Pass test with a broker and remote interface without remote worker.

Close #161

wusyong commented on src/remote_common.c:
```diff
@@ -186,7 +187,11 @@ bool wait_response_message(amqp_connection_state_t *conn,
         amqp_maybe_release_buffers(*conn);
-        amqp_simple_wait_frame(*conn, &frame);
+
+        struct timeval t = {10, 0}; /* RPC timeout: 10s*/
```
Just curious why 10 seconds is chosen. And maybe these could be macros?

ajblane replied:
It can be changed to other values to make sense and I don't know much about that. 
For consistent, MACRO does not be used.

jserv commented:
Don't change the lines related to `ddprintf`. Let's improve diagnosis message in another pull request. In this RP, we shall refactor.

jserv commented on src/remote_common.c:
```diff
@@ -186,7 +187,11 @@ bool wait_response_message(amqp_connection_state_t *conn,
-        amqp_simple_wait_frame(*conn, &frame);
+
+        struct timeval t = {10, 0}; /* RPC timeout: 10s*/
+        if(!die_on_error(amqp_simple_wait_frame_noblock(*conn, &frame, &t), "RPC timeout"))
```
Did you use `clang-format -i` to indent?

ajblane committed:
refactor: Change blocked RPC to non-blocked RPC

In the case of all remote workers which are not workable, blocked PRCs
not get RPC results until remote workers are workable. Therefore,
remote interface needs to change blocked RPC to non-blocked RPC with
10s. When time-out occurs, remote interface goes to run local dcurl.

Pass test with a broker and remote interface without remote worker.

Close #161
